### DISCUSSION TITLE
[QUESTION] Is it possible to get a build/nuget package without GPU and PDF support

### DISCUSSION THREAD

**Discussion by trampster** [mono/SkiaSharp#1610] [Ideas]
We have a Xamarin App targeting Android and iOS, we are trying to reduce the size of our deployable. SkiaSharp is the single biggest thing in our deployment. 

We don't need GPU or PDF support, so it would be good if we could get a build of SkiaSharp that doesn't have these things. I'm not sure how much space this will save, but if it is significant it could be quite an improvement for us.

**Comment by Gillibald** [comment]
SkiaSharp.dll -> 400KB
libSkiaSharp.so ~ 8MB
libSkiaSharp (iOS) ~ 21MB

Removing some code from the native bits will not change much. Looks like the iOS build statically links a lot into the binary.

**Comment by mattleibow** [comment]
I wonder what else we can remove that is taking up space. At 21 MB for ios means that each arch is about 5 MB.

I suppose we could flip bits all over, but is there a more scientific way to see which symbols are the ones pulling in the most bits?

It might be the external dependencies as well. I am linking it, but we might be able to see if we can dynamically link with libs in the NDK... But then we run into issues with compatibility.

An option would be to both dynamically and statically link for iOS and then pick dynamic for debug and static for release. I need to check, but I don't think much was saved when I did this a few months back...
